c2corg_api/views/document_delete.py

```diff
@@ -43,26 +42,52 @@ def validate_document(request, **kwargs):
         filter(Document.document_id == document_id).first()
 
     if not document:
-        raise HTTPNotFound('document not found')
+        request.errors.add(
+            'querystring',
+            '{}'.format(document_id),
+            'document not found')
+        request.errors.status = 404
+        return
 
     document_type = document.type
+    if document_type not in (WAYPOINT_TYPE,
+                             ROUTE_TYPE,
+                             OUTING_TYPE,
+                             IMAGE_TYPE,
+                             ARTICLE_TYPE,
+                             BOOK_TYPE,
+                             XREPORT_TYPE):
+        request.errors.add(
+            'querystring',
+            '{}'.format(document_id),
```

Comment:
`name` is usually the name of the field that contains some invalid data, and not the value of that field. In this case I think we could use `document_id`.